app/src/main/java/xyz/whereuat/whereuat/MainActivity.java

```diff
@@ -148,22 +140,34 @@ public void onReceive(Context context, Intent intent) {
         }
     }
 
-    public void addContact(View view){
+    public void addContactOnClick(View view){
         mMenu.close(true);
+        if (requestContactPermissions()) {
+            showContactsPhonebook();
+        }
+    }
+
+    private boolean requestContactPermissions() {
```

Comment:
Rename to `hasContactPermissions`. Functions that return booleans should usually be prefixed with `is` or `has` or something.